src/main/java/com/google/devtools/build/lib/worker/WorkerSpawnRunner.java

```diff
@@ -112,6 +112,14 @@ public String getName() {
     return "worker";
   }
 
+  private String getMultiplexPrefixName(boolean proxied) {
+    if (proxied) {
+      return "multiplex-" + getName();
+    } else {
+      return getName();
+    }
+  }
```

Comment:
This logic is duplicated in `WorkerKey`. Maybe it can be reused?

Reply:
Sounds reasonable. I think we have no choice but to keep the logic in `WorkerSpawnRunner` since there is no `WorkerKey` yet when we call the first `context.report()`. But in order to get rid of the duplicate logic, we will have to use a new member variable.

Some explanation of design choices:
1. We could replace the second `context.report()` by
```
context.report(ProgressStatus.EXECUTING, key.getWorkerTypeName());
```
But I like it to be consistent calling the same function `getNameForReporting()` in same file. What do you think?
2. It looks like `Spawns.supportsMultiplexWorkers(spawn)` has been called multiple times in `WorkerSpawnRunner`. We could store it in a variable in `exec()` and pass it all the way down to `actuallyExec()` and `execInWorker()`. It may look cleaner. What do you think?

Reply:
Have a static function `WorkerKey#makeWorkerTypeName(String name)` that builds the name, and then call it from wherever you need. No need for member fields.

Reply:
Thanks, I think it will work. Just to verify, would the function be `WorkerKey#makeWorkerTypeName(boolean proxied)` instead? Passing `String name` to the function doesn't make sense to me, would you mind to elaborate more?

Reply:
I assume `boolean` makes more sense, just updated. Let me know if you have different thought.